frontend/src/app/api/fetchers.ts

```diff
@@ -0,0 +1,72 @@
+import "server-only";
+
+import {
+  EndpointConfig,
+  fetchOpportunityEndpoint,
+  opportunitySearchEndpoint,
+} from "src/app/api/endpointConfigs";
+import {
+  createRequestBody,
+  createRequestUrl,
+  getDefaultHeaders,
+  HeadersDict,
+  JSONRequestBody,
+  sendRequest,
+} from "src/app/api/fetchHelpers";
+import { APIResponse } from "src/types/apiResponseTypes";
+import { OpportunityApiResponse } from "src/types/opportunity/opportunityResponseTypes";
+import { QueryParamData } from "src/types/search/searchRequestTypes";
+import { SearchAPIResponse } from "src/types/search/searchResponseTypes";
+
+import { cache } from "react";
+
+// returns a function which can be used to make a request to an endpoint defined in the passed config
+export function requesterForEndpoint<ResponseType extends APIResponse>({
```

Comment:
originally had this living elsewhere, but testing was much easier if I isolated it away from the other helper fns